Tests/SessionTests.swift

```diff
@@ -876,8 +888,10 @@ class SessionTestCase: BaseTestCase {
         waitForExpectations(timeout: timeout, handler: nil)
 
         // Then
+        XCTAssertEqual(handler.adaptCalledCount, 2)
         XCTAssertEqual(handler.adaptedCount, 2)
-        XCTAssertEqual(handler.retryCount, 2)
+        XCTAssertEqual(handler.retryCalledCount, 3)
```

Comment:
Can you briefly outline why these counts have changed? Is it due to changes in what we're counting as retries between 4 and 5?

Reply:
Absolutely. The reason why the retry counts have increased by one in almost all cases here is that AF5 now asks the retrier if it would like to retry in cases where response serialization fails. Since we didn't previously support that, most of these tests will see one more retry pass.

As an additional callout, I added `adaptCalledCount` and `retryCalledCount` to the already existing `adaptCount` and `retryCount` properties. The `Called` variants track if the method was ever called, where the non-Called variants only track whether the adapt or retry successfully execute.